services/dynamic/dynamic-xml.service.js

```diff
@@ -70,22 +70,28 @@ export default class DynamicXml extends BaseService {
 
   static defaultBadgeData = { label: 'custom badge' }
 
-  transform({ pathExpression, buffer }) {
+  transform({ pathExpression, buffer, contentType = 'text/xml' }) {
     // e.g. //book[2]/@id
     const pathIsAttr = (
       pathExpression.split('/').slice(-1)[0] || ''
     ).startsWith('@')
 
     let parsed
     try {
-      parsed = new DOMParser().parseFromString(buffer, 'text/xml')
+      parsed = new DOMParser().parseFromString(buffer, contentType)
     } catch (e) {
       throw new InvalidResponse({ prettyMessage: e.message })
     }
 
     let values
     try {
-      values = xpath.select(pathExpression, parsed)
+      if (contentType === 'text/html') {
+        values = xpath
+          .parse(pathExpression)
+          .select({ node: parsed, isHtml: true })
```

Comment:
This is required so that we don't have to explicitly use a namespace on every selector for HTML documents

See

- https://github.com/xmldom/xmldom/issues/377
- https://github.com/goto100/xpath/issues/27

for more detail